### DISCUSSION TITLE
Alias from main query not recognized in subquery join `symbol m.id not found`

### DISCUSSION THREAD

**Discussion by CanRau** [planetscale/discussion#425] [Q&A] [answered]
I've got a query, here's a very simplified and contrived example

```
SELECT
  id,
  (
    SELECT JSON_ARRAYAGG(JSON_OBJECT("id", id))
    FROM other_table AS ot
      INNER JOIN third_table AS tt ON tt.main_id = m.id <---- this `m.id` is erroring 😨
    WHERE m.status = 'active' <--- works
  ) AS subquery
FROM main_table AS m
WHERE m.id = 3;
```
which works great in my local MySQL Docker instance running the same version 8.0.23 as my pscale db

just ran it for the first time against pscale and get `symbol m.id not found`.

As shown in the example I use the alias `m` throughout the query in various subqueries but narrowed it down to this specific case when used in a subqueries join, replacing it with an actual id like `ON tt.main_id = 3` works as expected.

Am I missing anything? I tried searching here on GH in the planetscale org across repos but couldn't find anything.

Many of my queries depend on this and now I'm a little lost as deadline for migrating our data and queries to Planetscale is Friday 😳😅

**Comment by orware** [comment, marked as answer]
Hi @CanRau,

I don't believe the team will be able to make your Friday deadline for a potential fix unfortunately, as more than likely this will need to be replicated and then passed on as an issue to our Vitess team for review and bug fixing which normally takes a bit of time, unless a different workaround ends up being available we can utilize.

I'll keep you posted on the progress I'm able to make in replicating the issue!

Also, if you'd like to, we can also work a bit more closely with you via ticket over at https://support.planetscale.com/ if you'd like to send one in I can keep an eye out for it.

**Reply by CanRau** [reply]
Oh I see, I hoped for it not be a bug or similar 🥲
Thanks a lot for getting back so quickly Omar 🙏

So for now I see what to do in the meantime. Just sent in a support ticket `#4234`

**Reply by orware** [reply]
Thank you @CanRau!

I'm going to start replying there so we can continue the conversation.
